server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -1372,6 +1372,11 @@ public TSStatus insertRecords(TSInsertRecordsReq req) {
           allCheckSuccess = false;
         }
         insertRowsPlan.addOneInsertRowPlan(plan, i);
+      } catch (IoTDBException e) {
+        allCheckSuccess = false;
+        insertRowsPlan
+            .getResults()
+            .put(i, onIOTDBException(e, "inserting records", e.getErrorCode()));
```

Comment:
Better to move all these "inserting records" to a IoTDBConstant or a n Enum class